python/hsfs/core/arrow_flight_client.py

```diff
@@ -64,7 +64,18 @@ def __init__(self):
     def _initialize_connection(self):
         self._client = client.get_instance()
 
-        host_url = "grpc+tls://flyingduck.service.consul:5005"
+        if isinstance(self._client, client.external.Client):
+            try:
+                external_domain = self._variable_api.get_loadbalancer_external_domain()
```

Comment:
If i understood the code correctly, you are doing this for dev machines where flyingduck is on the same node. However, on managed this would never happen, so instead i would throw an exception that "external access flyingduck is not supported" is the variable is empty

Reply:
Indeed that's the reason. Could you point me to where I can find out whether the client externally connects to a dev machine vs. a cloud instance?

Reply:
fixed as discussed on slack